wa/instruments/perf.py

```diff
@@ -134,5 +182,94 @@ def update_output(self, context):
                                 metric = '{}_{}'.format(label, match.group(3))
                                 context.add_metric(metric, count, classifiers=classifiers)
 
-    def teardown(self, context):
-        self.collector.reset()
+    @staticmethod
+    def _process_perf_record_output(context, outdir):
```

Comment:
The methods below need to be refactored --  please reduce the level of nesting and make sure lines are of reasonable length (as a rule of thumb --  the entire line should be viewable in GitHub without scrolling.

Reply:
Refactored output processing related methods. Thanks, i'll make sure to adhere to this in future.